catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/Source.view.js

```diff
@@ -137,7 +143,27 @@ define([
             template: 'sourceList',
             itemView: SourceView.SourceRow,
             emptyView: EmptyView.sources,
-            itemViewContainer: 'tbody'
+            itemViewContainer: 'tbody',
+
+            initialize: function() {
+                this.hidePageLoading();
+                this.showSourcesLoading();
+            },
+            collectionEvents: {
+                "request": "showSourcesLoading",
+                "sync": "hideSourcesLoading"
+            },
+            showSourcesLoading: function() {
+                this.$el.addClass('hide');
+                $("#sources-loading").removeClass('hide');
+            },
+            hideSourcesLoading: function() {
+                  $("#sources-loading").addClass('hide');
```

Comment:
Worked with @andrewkfiedler to cleanup the code some.  He thought this approach was easiest to implement and easiest to understand.